### PR TITLE
Resolve #1161, #1148, #1178, #1197

### LINKED ISSUE #1161: Receipt security answer should not be base58 encoded

### PR DESCRIPTION
**Pull request by InVincible2016**

### CODE REVIEW

File: src/components/AppBarComponent.jsx [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+                      {this.renderChainsfrLogo()}
+                    </Hidden>
+                  ) : (
+                    this.renderChainsfrLogo()
+                  )}
+                </Grid>
+                <Hidden only={['xs', 'sm']}>
+                  {profile.isAuthenticated &&
+                    cloudWalletConnected &&
+                    ([path.receive, path.transfer].includes(location.pathname) ? (
```
Comment (timothywangdev) [inline src/components/AppBarComponent.jsx]: rebase and add path.directTransfer

Comment (timothywangdev) [reply src/components/AppBarComponent.jsx]: also add stepper for directTransfer

File: src/components/NavDrawerComponent.jsx [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+                  location.pathname === btn.path ? classes.NaviBtnListItemMobileActive : undefined
+                }
+              >
+                <Button
+                  className={
+                    location.pathname === btn.path ? classes.NaviBtnMobileActive : classes.NaviBtn
+                  }
+                  component={Link}
+                  to={btn.path}
+                  onClick={() => {}}
```
Comment (timothywangdev) [inline src/components/NavDrawerComponent.jsx]:
```suggestion
```

File: src/components/NavDrawerComponent.jsx [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+                >
+                  {btn.name}
+                </Button>
+              </Box>
+            </ListItem>
+          ))}
+          <ListItem style={{ marginTop: 'auto' }}>
+            <Button onClick={() => {}} id='logout' className={classes.logoutBtn}>
```
Comment (timothywangdev) [inline src/components/NavDrawerComponent.jsx]: logout has not functionality

File: src/containers/AppBarContainer.jsx [hunk 4]
```diff
@@ -15,17 +15,27 @@ class NavBarContainer extends Component {
   }
 
   render () {
-    let { onLogout, profile, location, cloudWalletConnected } = this.props
+    let { onLogout, profile, location, cloudWalletConnected, handleDrawerToggle } = this.props
     const urlParams = queryString.parse(location.search)
     let step = parseInt(urlParams.step) || 0
+    const navigatabled = ![
```
Comment (timothywangdev) [inline src/containers/AppBarContainer.jsx]: typo navigatable

File: src/components/AppBarComponent.jsx [hunk 5]
```diff
@@ -0,0 +1,240 @@
+import React, { Component } from 'react'
```
Comment (timothywangdev) [inline src/components/AppBarComponent.jsx]: missing flow

File: src/components/NavDrawerComponent.jsx [hunk 6]
```diff
@@ -0,0 +1,264 @@
+import React from 'react'
```
Comment (timothywangdev) [inline src/components/NavDrawerComponent.jsx]: missing flow

File: src/components/AppBarComponent.jsx [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+  render () {
+    const {
+      classes,
+      profile,
+      disabled,
+      location,
+      step,
+      cloudWalletConnected,
+      handleDrawerToggle,
+      navigatabled
```
Comment (timothywangdev) [inline src/components/AppBarComponent.jsx]: comment on this param in flow

File: src/containers/AppBarContainer.jsx [hunk 8]
```diff
@@ -15,17 +15,27 @@ class NavBarContainer extends Component {
...
+      path.transfer,
+      path.cancel,
+      path.receipt,
+      path.receive,
```
Comment (timothywangdev) [inline src/containers/AppBarContainer.jsx]: add path.directTransfer

File: src/containers/NavDrawerContainer.jsx [hunk 9]
```diff
@@ -0,0 +1,37 @@
+import React, { PureComponent } from 'react'
```
Comment (timothywangdev) [inline src/containers/NavDrawerContainer.jsx]: missing flow

File: src/components/AppBarComponent.jsx (hunk 1 above)
Comment (InVincible2016) [reply src/components/AppBarComponent.jsx]: will do in a separate commit

File: src/actions/walletActions.js [hunk 10]
```diff
@@ -105,7 +105,7 @@ async function _getCloudWallet () {
     throw new Error(WalletErrors.drive.walletNotExist)
   }
 
-  let accountDataList = Base64.decode(walletFile.accounts)
+  let accountDataList = JSON.parse(Base64.decode(walletFile.accounts))
```
[the remark this answers is not in the document]
Comment (InVincible2016) [reply src/actions/walletActions.js]: This itself has its own issue #1161

**Pull request merged**